File: java/src/main/java/pl/kkurczewski/deser/greet/GreetingController.java
package pl.kkurczewski.deser.greet;

import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;
import pl.kkurczewski.deser.greet.dto.Greeting;

@RestController
public class GreetingController {

    @PostMapping("/greet")
    public Greeting greeting(@RequestBody Greeting greeting) {
        return greeting;
    }
}

File: kotlin/src/main/kotlin/pl/kkurczewski/deser/greet/GreetingController.kt
package pl.kkurczewski.deser.greet

import pl.kkurczewski.deser.greet.dto.Greeting
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class GreetingController {
    @PostMapping("/greet")
    fun greeting(@RequestBody greeting: Greeting) = greeting
}